main.py

```diff
@@ -1,18 +1,183 @@
 import os
+import json
+import hmac
+import hashlib
+import google.cloud.logging
+import logging
+import jsonschema
+import requests
+import pandas as pd
+import numpy as np
+import io
 
 from flask import escape
-from google.cloud import pubsub_v1
+from jsonschema import validate
+from confluent_kafka import Producer, KafkaError
 
+delivered_records = 0
 
-def publish_message(topic_name, msg):
-    publisher = pubsub_v1.PublisherClient()
-    future = publisher.publish(topic_name, msg)
-    future.result()
+client = google.cloud.logging.Client()
+client.setup_logging()
 
+# Describe what kind of json you expect.
+releasePayloadSchema = {
+  "type": "object",
+  "properties": {
+    "action": {"type": "string"},
+    "release": {
+      "type": "object",
+      "properties": {
+        "tag_name": {"type": "string"},
+        "name": {"type": "string"},
+        "draft": {"type": "boolean"},
+        "prerelease": {"type": "boolean"},
+        "created_at": {"type": "string", "format": "date-time"},
+        "published_at": {"type": "string", "format": "date-time"},
+        "assets": {
+          "type": "array",
+          "items": {
+            "type":"object",
+            "properties": {
+              "name": {"type": "string"},
+              "browser_download_url": {"type": "string"}
+            },
+            "required": ["name","browser_download_url"]
+          }
+        }
+      },
+      "required": ["name","draft","prerelease","published_at","assets"]
+    },
+    "repository": {
+      "type": "object",
+      "properties": {
+        "name": {"type": "string"},
+        "full_name": {"type": "string"}
+      },
+      "required": ["name","full_name"]
+    }
+  },
+  "required": ["action","release","repository"]
+}
 
-def pubsub_queue():
-    return os.environ.get('PUBSUB_QUEUE')
+def validate_signature(request):
 
+    # logging.info(json.dumps(dict(request.headers), indent=4))
+    signature_header = request.headers.get('X-Hub-Signature-256')
+    sha_name, github_signature = signature_header.split('=')
+    if sha_name != 'sha256':
+      logging.error('X-Hub-Signature in payload headers was not sha256=****')
```

Comment:
You've got a mix of 4 space and 2 space indents -- please use 4 spaces for all indents. Python will get sad if someone changes this later and uses 4 space indents in the wrong spot 🐍

Reply:
done